distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1085,90 +1099,144 @@ public void testDownloadFromFederatedCswSource() throws Exception {
+                        .size() == 1);
+        expect("Waiting for activities").within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(ACTIVITIES_CHANNEL)
+                        .size() == 2);
+
+        assertThat(cometDClient.getAllMessages()
+                .size(), is(3));
+
         List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
         assertThat(notifications.size(), is(1));
-        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)), filename, "Resource retrieval completed, 8 bytes retrieved. ", "complete");
-       
-        List<String> activities = cometDClient.getMessagesInAscOrder(ACTIVITIES_CHANNEL); 
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                "Resource retrieval completed, 8 bytes retrieved. ",
+                "complete");
+
+        List<String> activities = cometDClient.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
         assertThat(activities.size(), is(2));
-        CometDMessageValidator.verifyActivity(JsonPath.from(activities.get(0)), filename, "Resource retrieval started. ", "STARTED");
-        CometDMessageValidator.verifyActivity(JsonPath.from(activities.get(1)), filename, "Resource retrieval completed, 8 bytes retrieved. ", "COMPLETE");
+        CometDMessageValidator.verifyActivity(JsonPath.from(activities.get(0)),
+                filename,
+                "Resource retrieval started. ",
+                "STARTED");
+        CometDMessageValidator.verifyActivity(JsonPath.from(activities.get(1)),
+                filename,
+                "Resource retrieval completed, 8 bytes retrieved. ",
+                "COMPLETE");
     }
 
     /**
      * Tests that if the endpoint disconnects twice, the retrieval retries both times
+     *
      * @throws Exception
      */
     @Test
     public void testRetrievalReliablility() throws Exception {
         cometDClient = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL, ACTIVITIES_CHANNEL));
         String filename = "product2.txt";
 
+        cswServer.setMetacardId(CSW_STUB_RESOURCE_ID + 2);
         cswServer.whenHttp()
                 .match(Condition.get("/services/csw"),
                         Condition.parameter("request", "GetRecordById"))
-                .then(getCswRetrievalHeaders(filename), chunkedContentWithHeaders(STUB_SERVER_TEST_STRING, Duration.ofMillis(200), 2));
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(STUB_SERVER_TEST_STRING,
+                                Duration.ofMillis(200),
+                                2));
 
         String restUrl =
-                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
+                REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID + 2
```

Comment:
I see the `+2` here and the `+3` down below. Can we extract those into named constants for clarity?